MediaBrowser.Providers/Plugins/TheTvdb/TvdbClientManager.cs

```diff
@@ -21,67 +22,97 @@ public class TvdbClientManager
         private const string DefaultLanguage = "en";
 
         private readonly IMemoryCache _cache;
-        private readonly TvDbClient _tvDbClient;
-        private DateTime _tokenCreatedAt;
+
+        /// <summary>
+        /// TvDbClients per language.
+        /// </summary>
+        private readonly ConcurrentDictionary<string, TvDbClient> _tvDbClients = new ConcurrentDictionary<string, TvDbClient>();
+
+        private readonly SemaphoreSlim _tokenUpdateLock = new SemaphoreSlim(1, 1);
+        private DateTime _tokenUpdatedAt;
 
         public TvdbClientManager(IMemoryCache memoryCache)
         {
             _cache = memoryCache;
-            _tvDbClient = new TvDbClient();
         }
 
-        private TvDbClient TvDbClient
+        private async Task<TvDbClient> GetTvDbClient(string language)
         {
-            get
+            var normalizedLanguage = TvdbUtils.NormalizeLanguage(language) ?? DefaultLanguage;
+
+            var tvDbClient = _tvDbClients.GetOrAdd(normalizedLanguage, (key) => new TvDbClient()
             {
-                if (string.IsNullOrEmpty(_tvDbClient.Authentication.Token))
-                {
-                    _tvDbClient.Authentication.AuthenticateAsync(TvdbUtils.TvdbApiKey).GetAwaiter().GetResult();
-                    _tokenCreatedAt = DateTime.Now;
-                }
+                AcceptedLanguage = key
+            });
 
-                // Refresh if necessary
-                if (_tokenCreatedAt < DateTime.Now.Subtract(TimeSpan.FromHours(20)))
+            // First time authenticating
+            if (string.IsNullOrEmpty(tvDbClient.Authentication.Token))
+            {
+                try
                 {
-                    try
+                    await _tokenUpdateLock.WaitAsync().ConfigureAwait(false);
+
+                    if (string.IsNullOrEmpty(tvDbClient.Authentication.Token))
                     {
-                        _tvDbClient.Authentication.RefreshTokenAsync().GetAwaiter().GetResult();
+                        await tvDbClient.Authentication.AuthenticateAsync(TvdbUtils.TvdbApiKey).ConfigureAwait(false);
+                        _tokenUpdatedAt = DateTime.UtcNow;
```

Comment:
Might have issues with auth tokens if the clients aren't created at the same time

Reply:
Good call out. I refactored it a bit so there's a separate token time/lock per client